sdk/core/perf/inc/azure/perf/random_stream.hpp

```diff
@@ -0,0 +1,53 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+/**
+ * @file
+ * @brief A random stream of any specific size. Useful for test cases.
+ *
+ */
+
+#pragma once
+
+#include <azure/core/io/body_stream.hpp>
+
+#include <memory>
+#include <random>
+
+namespace Azure { namespace Perf {
+
+  /**
+   * @brief A random stream of any specific size. Useful for test cases.
+   *
+   */
+  class RandomStream {
+  private:
+    /**
+     * @brief Wraps a stream and keep reading bytes from it by rewinding it until some length.
+     *
+     * @note Enables to create a stream with huge size by re-using a small buffer.
+     *
+     */
+    class CircularStream : public Azure::Core::IO::BodyStream {
+    private:
+      std::unique_ptr<std::vector<uint8_t>> m_buffer;
+      size_t m_length;
+      size_t m_totalRead = 0;
+      Azure::Core::IO::MemoryBodyStream m_memoryStream;
```

Comment:
Just a question (maybe for future iterations): I wonder if this should be any BodyStream, or one that is passed-in from the ctor.

Reply:
In other languages, the CircularStream is just a wrapper on top of any BodyStream which rewinds the inner body stream as a way to simulate a bigger bodyStream.  It does not care about where the real data is and only rely on the innerBodyStream.

For C++, we don't have the luxury of allocating memory in the heap and then create a bodyStream on top of it and foget about it. We need to control who will clean that memory (we don't like GC :) .  So, I make the CircularStream in CPP to own the memory.

When the circualStream is created, the ctor takes only the size. Then it allocates that size in the heap and builds a MemoryBodyStream on top of that memory. As soon as the CircularStream lives, so the memory, and as soon as the CircularStream is destroyed, so the memory.

The RandomStream creates a CircularStream and returns it as its base class :).   This is que equivalent of the way it works on other languages.